string file_name

---

string message

bool result